### PR TITLE
switch to yarn and create TS version of setup.js

### PR DESCRIPTION
**Pull request by bdmason**
The use of `yarn` helped deal with some issues with nested dependencies.

There are a few `TODO_CHECK` comments in the code. A couple are quite minor, but there are 3 `@ts-ignore` comments where removing the `@ts-ignore` highlights issues, however these issues don't effect runtime. When running `npm run setup`, the code in question never runs as `setupRegistrar` exist on the return on line 292, straight after `console.log('registrar already existed, ignore.');`.

It's about as far as I can take it without understanding the domain a little better. Perhaps writing a bunch of empty tests e.g.
```js
describe('setup.ts', () => {
  it('Does A', () => {})
  it('Does B', () => {})
  it('Does C', () => {})
})
```
would help, as I feel like I'm misunderstanding the purpose (or how to use) the file.

All that has been achieved so far is to make running `ts-node setup.ts` behave the same as running `node setup.js`.

_Please add any developers that know the repository to the reviewers list_

### CODE REVIEW

File: setup.ts [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+
+    // TODO_CHECK is this being used?
+    private readonly unsubscribeEventListener: null;
+
+    /**
+     * A wrapped APIs for block chain
+     * @constructor
+     */
+    constructor(config: Config) {
+        // if (!self) {
```
Comment (chenzongxiong) [inline setup.ts:62]: remove these comment lines.

File: setup.ts [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+        }
+
+        /**
+         * set registrar fee and query results
+         */
+        const fee = DEFAULT_REGISTRAR_FEE;
+        await this.identitySetFee(registrarAccount, regIndex, fee);
+        await sleep(DEFAULT_SLEEP_INTERVAL);
+        await this.identityRegistrars();
+        await this.proxyAddProxy(registrarAccount, this.eve.address);
```
Comment (chenzongxiong) [inline setup.ts:366]: Consider this line and line 388, it would be duplicated. We can remove this line. 

Let's do `account proxy` in line 387 - line 390

**Comment by bdmason** [comment]
@chenzongxiong do you want me to update the staging workflow yet or would you rather leave it on the JS version for now?

**Comment by chenzongxiong** [comment]
> @chenzongxiong do you want me to update the staging workflow yet or would you rather leave it on the JS version for now?

I'd like to migrate to `setup.ts` directly.

**Review by chenzongxiong — APPROVED** [review]

**Pull request merged**